src/event_listeners.jl

```diff
@@ -0,0 +1,139 @@
+        println("done.")
+
+        GitHub.post_status(event, sha, GitHub.SUCCESS,
+                           auth = auth, context = "Benchmarker",
+                           description = "Benchmarks complete!")
+
+        return HttpCommon.Response(200)
+    end
+
+    # Start the server on port 8000
+    GitHub.run(listener, 8000)
+
+"""
+immutable EventListener
+    server::HttpServer.Server
+    function EventListener(handle;
+                           auth::Authorization = AnonymousAuth(),
+                           secret = nothing,
+                           events = nothing,
+                           repos = nothing,
+                           forwards = nothing)
+        if !(isa(forwards, Void))
+            forwards = map(HttpCommon.URI, forwards)
+        end
+
+        server = HttpServer.Server() do request, response
+            try
+                if !(isa(secret, Void)) && !(is_valid_secret(request, secret))
+                    return HttpCommon.Response(400, "invalid signature")
+                end
+
+                if !(isa(events, Void)) && !(is_valid_event(request, events))
+                    return HttpCommon.Response(400, "invalid event")
+                end
+
+                payload = Requests.json(request)
+
+                if !(isa(repos, Void)) && !(is_valid_repo(payload, repos))
+                    return HttpCommon.Response(400, "invalid repo")
+                end
+
+                if !(isa(forwards, Void))
+                    for address in forwards
+                        Requests.post(address,
+                                      UTF8String(request.data),
+                                      headers=request.headers)
+                    end
+                end
+
+                event = Event(request, payload)
+
+                return handle(event, auth)
+            catch err
+                println("SERVER ERROR: $err")
+                return HttpCommon.Response(500)
+            end
+        end
+
+        server.http.events["listen"] = port -> begin
+            println("Listening for GitHub events sent to $port;")
+            println("Whitelisted events: $(isa(events, Void) ? "All" : events)")
+            println("Whitelisted repos: $(isa(repos, Void) ? "All" : repos)")
+        end
+
+        return new(server)
+    end
+end
+
+function Base.run(listener::EventListener, args...; kwargs...)
+    return HttpServer.run(listener.server, args...; kwargs...)
+end
+
+########################
+# Validation Functions #
+########################
+
+has_sig_header(request::HttpCommon.Request) = haskey(request.headers, "X-Hub-Signature")
+sig_header(request::HttpCommon.Request) = request.headers["X-Hub-Signature"]
+
+function is_valid_secret(request::HttpCommon.Request, secret::AbstractString)
+    if has_sig_header(request)
+        payload_string = UTF8String(request.data)
+        secret_sha = "sha1="*Nettle.hexdigest("sha1", secret, payload_string)
```

Comment:
Splitting off a v1.0. branch to support 0.3 seems perfectly reasonable, and I think all this looks great (though switching to MbedTLS still seems like a good idea). Also (feel like I should say) sorry again for being so out of touch -- you are all cool, competent people and should feel free to merge stuff and not be held back